examples/energy-management-app/esp32/main/main.cpp

```diff
@@ -116,23 +119,48 @@ chip::Credentials::DeviceAttestationCredentialsProvider * get_dac_provider(void)
 
 } // namespace
 
+EVSEManufacturer * EnergyEvse::GetEvseManufacturer()
+{
+    return gEvseManufacturer;
+}
+
 void ApplicationInit()
 {
-    if ((gDelegate == nullptr) && (gInstance == nullptr) && (gEvseManufacturer == nullptr))
+    if ((gDEMDelegate == nullptr) && (gDEMInstance == nullptr))
```

Comment:
It might be worth aligning to the Linux\main.cpp here since there are more clusters being added here (including splitting out the CHIP_ERROR DeviceEnergyManagementInit(), DeviceEnergyManagementShutdown()
and CHIP_ERROR EnergyEvseInit()/CHIP_ERROR EnergyEvseShutdown()
and CHIP_ERROR EVSEManufacturerInit() / CHIP_ERROR EVSEManufacturerShutdown()

There is another PR to add in Electrical Power and Energy measurement clusters in a similar way.

Also using the more up to date make_unique style means you don't call new/delete and it makes the clean-up code neater.